api/internal/localizer/locloader.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2022 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package localizer
+
+import (
+	"log"
+	"path/filepath"
+
+	"sigs.k8s.io/kustomize/api/ifc"
+	"sigs.k8s.io/kustomize/api/internal/git"
+	"sigs.k8s.io/kustomize/api/loader"
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+const dstPrefix = "localized"
+
+// LocLoader is the Loader for kustomize localize. It Loads under localize constraints.
+type LocLoader struct {
+	fSys filesys.FileSystem
+
+	// directory that bounds target's local references, empty string if target is remote
+	scope filesys.ConfirmedDir
+
+	// localize destination
+	newDir filesys.ConfirmedDir
+
+	// loader at LocLoader's current directory
+	ifc.Loader
+
+	// whether LocLoader and all its ancestors are the result of local references
+	local bool
+}
+
+// ValidateLocArgs validates the arguments to kustomize localize.
+// On success, ValidateLocArgs returns a LocLoader at targetArg and the localize destination directory; otherwise, an
+// error.
+func ValidateLocArgs(targetArg string, scopeArg string, newDirArg string, fSys filesys.FileSystem) (*LocLoader, filesys.ConfirmedDir, error) {
+	// check earlier to avoid cleanup
+	spec, err := git.NewRepoSpecFromURL(targetArg)
+	if err == nil && spec.Ref == "" {
+		return nil, "", errors.Errorf("localize remote root '%s' missing ref query string parameter", targetArg)
+	}
+
+	// for security, should enforce load restrictions
+	ldr, err := loader.NewLoader(loader.RestrictionRootOnly, targetArg, fSys)
+	if err != nil {
+		return nil, "", errors.WrapPrefixf(err, "unable to establish localize target '%s'", targetArg)
+	}
+
+	scope, err := validateScope(scopeArg, targetArg, ldr, fSys)
+	if err != nil {
+		_ = ldr.Cleanup()
+		return nil, "", errors.WrapPrefixf(err, "invalid localize scope '%s'", scopeArg)
+	}
+
+	newDir, err := validateNewDir(newDirArg, ldr, spec, fSys)
+	if err != nil {
+		_ = ldr.Cleanup()
+		return nil, "", errors.WrapPrefixf(err, "invalid localize destination '%s'", newDirArg)
+	}
+
+	return &LocLoader{
+		fSys:   fSys,
+		scope:  scope,
+		newDir: newDir,
+		Loader: ldr,
+		local:  scope != "",
+	}, newDir, nil
+}
+
+// Scope returns the localize scope
+func (ll *LocLoader) Scope() string {
```

Comment:
Where would the scope be used outside of `LocLoader.Load`?

Reply:
In the next layer, `localizer`, the `scope` is used to determine the directory in `newDir` that corresponds to `target`.